### PR TITLE
fix[upa]: fail gracefully for point data which cannot be decompressed

### PR DESCRIPTION
**Pull request by dtebbs**
Gracefully handle the case where group points cannot be decompressed.

- [x] Set a valid title
- [x] Use branch-name `[<ref>-]<feature-description>`
- [x] Brief description of the changes
- [x] Create matching PRs in dependent repos

### CODE REVIEW

File: upa/src/sdk/pointCompression.ts [hunk 1]
```diff
@@ -51,6 +54,10 @@ export function decompressG1Point(compressedPoint: CompressedG1Point): G1Point {
   const x3: bigint = FQ.mul(x2, x);
   const x3PlusB: bigint = FQ.add(x3, BN254_G1_B);
   const y: bigint = FQ.sqrt(x3PlusB);
+  if (!y) {
+    return undefined;
+  }
+
   assert("bigint" == typeof y, "sqrt failed");
```
Comment (dysfunctor) [inline upa/src/sdk/pointCompression.ts]: Probably can take this assert out (to take out the crashing path)- I think y could only be have been null or bigint.

File: upa/src/sdk/pointCompression.ts [hunk 2]
```diff
@@ -79,6 +89,13 @@ export function decompressG2Point(compressedPoint: CompressedG2Point): G2Point {
   const x3: [bigint, bigint] = FQ2.mul(x2, x);
   const x3PlusB: [bigint, bigint] = FQ2.add(x3, [BN254_G2_B0, BN254_G2_B1]);
   const y: [bigint, bigint] = FQ2.sqrt(x3PlusB);
+
+  if (!y) {
+    return undefined;
+  }
+
+  assert("bigint" == typeof y[0], "sqrt failed");
```
Comment (dysfunctor) [inline upa/src/sdk/pointCompression.ts]: Similar comment on these asserts

**Review by dysfunctor — APPROVED** [review]
There's one more use of `Submission.fromTransactionReceipt` to undefined-check in the get-submission command

**Review by akileshtangella — APPROVED** [review]

File: upa/src/sdk/pointCompression.ts (hunk 2 above)
Comment (dtebbs) [reply upa/src/sdk/pointCompression.ts]: done

File: upa/src/sdk/pointCompression.ts (hunk 1 above)
Comment (dtebbs) [reply upa/src/sdk/pointCompression.ts]: done

**Pull request merged**